### PR TITLE
Add some clarification to NVME beta instructions

### LINKED ISSUE #1939: NVME guide assumes that the user is familiar with usbboot
@peterharperuk I've seen a number of forum posts where the user is running usbboot on a CM4 as a way of enabling NVME booting. The documentation here does not make it clear that usbboot/rpiboot must be run on a separate computer Pi/PC/MAC in order to update the CM4 boot EEPROM.

### PR DESCRIPTION
**Pull request by peterharperuk**
Add a bit more detail about running rpiboot from another PC
Fixes #1939

### CODE REVIEW

**Comment by lurch** [comment]
Does this look good to you? @pelwell @timg236

File: hardware/raspberrypi/bootmodes/nvme.md [hunk 1]
```diff
@@ -21,13 +21,17 @@ To boot from NVMe you need pre-release versions of the bootloader, VideoCore fir
 
 ### Bootloader
 
-You need to use `rpiboot` to update the CM4 bootloader. Instructions for building `rpiboot` and configuring the IO board to switch the ROM to usbboot mode are in the [usbboot Github repository](https://github.com/raspberrypi/usbboot).
+You need to use `rpiboot` to update the CM4 bootloader to use the beta bootloader. If you are using a CM4 with an eMMC then you must also change the BOOT_ORDER to give NVMe priority, otherwise the CM4 will continue to boot from eMMC.
```
Comment (pelwell) [inline hardware/raspberrypi/bootmodes/nvme.md:24]: I think it's worth mentioning _here_ that rpiboot must be run on another PC here, perhaps removing some of us duplication from the following paragraphs.
E.g. "Another computer running rpiboot must be used to update the EEPROM on a CM4".

**Comment by timg236** [comment]
No objection, although it seems document rpiboot again here when we already have the CM4 EMMC flashing guide.

File: hardware/raspberrypi/bootmodes/nvme.md [hunk 2]
```diff
@@ -21,13 +21,17 @@ To boot from NVMe you need pre-release versions of the bootloader, VideoCore fir
...
-If you are using a CM4 with an eMMC then you must change the BOOT_ORDER to give NVMe priority, otherwise the CM4 will continue to boot from eMMC. For example:
+Instructions for building `rpiboot` and configuring the IO board to switch the ROM to usbboot mode are in the [usbboot Github repository](https://github.com/raspberrypi/usbboot).
+
+To use `rpiboot` you need another PC (a Pi, PC or Mac), that is capable of building the `rpiboot` binary. This other PC is then connected to the CM4 IO board USB slave socket and you have to fit a jumper to the J2 header on the IO board to disable eMMC boot. Then when `rpiboot` is executed it communicates with the CM4 to reprogram the eeprom. You can't change the eeprom on a CM4 from the device itself.
+
+For example run the following from another PC connected to the CM4 USB slave connector:
 
 ```
 cd usbboot/nvme
-sed -i 's/\(BOOT_ORDER=.*\)6\(.*\)/\1\26/' boot.conf
```
Comment (timg236) [inline hardware/raspberrypi/bootmodes/nvme.md:30]: Get rid of sed. It's sufficient to tell people to just change the boot-order

**Comment by timg236** [comment]
Actually. Reading the full doc it only needs to contain the "Required hardware", "Required software" sections plus links to EMMC flashing guide and bootloader config.

The rest of the stuff is an artefact of the infrequent APT releases and the experience with USB MSD was that it's not worth writing custom guides for patching firmware releases for betas.

**Comment by peterharperuk** [comment]
I think referencing the eMMc update guide would be confusing, and I don't think it's that obvious how you update an NVME disk to include the latest firmware?

**Comment by timg236** [comment]
> I think referencing the eMMc update guide would be confusing, and I don't think it's that obvious how you update an NVME disk to include the latest firmware?

That would just be to say. You need at least EEPROM ver X with the BOOT_ORDER enabled and here are the normal instructions to do that.

It's probably worth having something explaining how to flash an image to an NVMe drive. However, maintaining official docs that say you have to manually rpi-update or copy firmware files about just leads to stale documentation. I think in future beta docs should just be a forum post until code eventually makes it into APT.

**Comment by peterharperuk** [comment]
Shall we just not bother? This is going to change again soon when the beta goes live.

**Comment by timg236** [comment]
> Shall we just not bother? This is going to change again soon when the beta goes live.

That also sounds ok. Probably for future beta stuff we should just have a sticky post on the forums, it's less effort to maintain and it's more obviously dev/beta stability.

**Comment by aallan** [comment]
I'm going to be freezing the current Markdown-based repo for PRs on Monday morning, see #1911 for details of what's going on. If this PR isn't merged before then it will unfortunately be deleted. Although we'll be happy to get a new PR against the new Asciidoc-based repo after that goes public — which will probably at the start of August.

**Comment by aallan** [comment]
Closing.

**Pull request closed without merging**